lnwallet/interface_test.go

```diff
@@ -1233,6 +1238,141 @@ func testSignOutputPrivateTweak(r *rpctest.Harness, w *lnwallet.LightningWallet,
 	}
 }
 
+func testReorgWalletBalance(r *rpctest.Harness, w *lnwallet.LightningWallet, t *testing.T) {
+	t.Logf("Running reorg wallet balance test")
+
+	// We first mine a few blocks to ensure any transactions still in the
+	// mempool confirm, and then get the original balance, before a
+	// reorganization that doesn't invalidate any existing transactions or
+	// create any new non-coinbase transactions. We'll then check if it's
+	// the same after the empty reorg.
+	_, err := r.Node.Generate(5)
+	if err != nil {
+		t.Fatalf("unable to generate blocks on passed node: %v", err)
+	}
+	// Give wallet time to catch up.
+	time.Sleep(time.Second)
+	// Send some money from the miner to the wallet
+	err = loadTestCredits(r, w, 20, 4)
+	if err != nil {
+		t.Fatalf("unable to send money to lnwallet: %v", err)
+	}
+	// Send some money from the wallet back to the miner.
+	// Grab a fresh address from the wallet to house this output.
+	minerAddr, err := r.NewAddress()
+	if err != nil {
+		t.Fatalf("unable to generate address for miner: %v", err)
+	}
+	script, err := txscript.PayToAddrScript(minerAddr)
+	if err != nil {
+		t.Fatalf("unable to create pay to addr script: %v", err)
+	}
+	output := &wire.TxOut{
+		Value:    1e8,
+		PkScript: script,
+	}
+	if _, err = w.SendOutputs([]*wire.TxOut{output}); err != nil {
+		t.Fatalf("unable to send outputs: %v", err)
+	}
+	_, err = r.Node.Generate(50)
+	if err != nil {
+		t.Fatalf("unable to generate blocks on passed node: %v", err)
+	}
+	// Give wallet time to catch up.
+	time.Sleep(time.Second)
+	// Get the original balance.
```

Comment:
Minor nit: missing a new line above here. And also below the call to `Generate`.